node-graph/gcore/src/text.rs

```diff
@@ -29,3 +42,142 @@ impl From<TextAlign> for parley::Alignment {
 		}
 	}
 }
+
+#[derive(Clone, DynAny)]
+pub struct Typography {
+	pub layout: Layout<()>,
+	pub font_family: String,
+	pub color: Color,
+	pub stroke: Option<(Color, f64)>,
+}
+
+impl fmt::Debug for Typography {
+	fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+		f.debug_struct("Typography")
+			.field("font_family", &self.font_family)
+			.field("color", &self.color)
+			.field("stroke", &self.stroke)
+			.finish()
+	}
+}
+
+impl PartialEq for Typography {
+	fn eq(&self, _other: &Self) -> bool {
+		true
+	}
+}
+
+impl Hash for Typography {
+	fn hash<H: Hasher>(&self, state: &mut H) {
+		self.layout.len().hash(state);
+	}
+}
+
+impl Typography {
+	pub fn to_vector(&self) -> Table<Vector> {
+		Table::new()
+	}
+}
+
+#[derive(Clone)]
+pub struct NewFontCacheWrapper(pub Arc<Mutex<NewFontCache>>);
+
+impl fmt::Debug for NewFontCacheWrapper {
+	fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+		f.debug_struct("font cache").finish()
+	}
+}
+
+impl PartialEq for NewFontCacheWrapper {
+	fn eq(&self, _other: &Self) -> bool {
+		log::error!("Font cache should not be compared");
+		false
+	}
+}
+
+unsafe impl dyn_any::StaticType for NewFontCacheWrapper {
+	type Static = NewFontCacheWrapper;
+}
+
+pub struct NewFontCache {
+	pub font_context: parley::FontContext,
+	pub layout_context: parley::LayoutContext<()>,
+	pub font_mapping: HashMap<Font, (String, parley::fontique::FontInfo)>,
+	pub hash: u64,
+}
+
+impl NewFontCache {
+	pub fn new() -> Self {
+		let mut new = NewFontCache {
+			font_context: parley::FontContext::new(),
+			layout_context: parley::LayoutContext::new(),
+			font_mapping: HashMap::new(),
+			hash: 0,
+		};
+
+		let source_sans_font = Font::new(SOURCE_SANS_FONT_FAMILY.to_string(), SOURCE_SANS_FONT_STYLE.to_string());
+		new.register_font(source_sans_font, SOURCE_SANS_FONT_DATA.to_vec());
+		new
+	}
+
+	pub fn register_font(&mut self, font: Font, data: Vec<u8>) {
+		match self.font_mapping.entry(font) {
+			Entry::Occupied(occupied_entry) => {
+				log::error!("Trying to register font that already is added: {:?}", occupied_entry.key());
```

Comment:
I don't think this should be treated as an error. Errors should be for serious issues in the code which might threaten the editor stability, this is not one of these cases